kaggle_cli/submit.py

```diff
@@ -129,29 +130,38 @@ def take_action(self, parsed_args):
             os.remove(target_name)
 
     @staticmethod
+    def _make_archive_name(original_file_path):
+        # if original name already has a suffix (csv,txt,etc), remove it
+        extension_pattern = r'(^.+)\.(.+)$'
+
+        # file may be in another directory
+        original_basename = os.path.basename(original_file_path)
+
+        if re.match(extension_pattern,original_basename):
+            archive_name = re.sub(extension_pattern,r'\1.zip',original_basename)
+        else:
+            archive_name = original_basename+".zip"
+
+        # this is used to prevent caching issues
```

Comment:
I still can't understand why it has caching issues. Can you explain the scenario?

Reply:
I'm not sure about this but I think that when we upload two submissions (different files) with the same same (say, "mysubmission.csv"), this may trigger some form of caching in Kaggle 

(They may only check the file name and, if the file name is something they have seen before, they may not even read the file).

Again, I'm not sure about this but since caching is done by many people on many websites, I though it would be a safe thing to do at little cost (just add a couple characters to the file name).

But I agree it's not essential.

Reply:
Caching usually applied on read operation, caching a file upload would be considered a bug.

But I just tested if Kaggle caches the file? The answer is no.

I uploaded the benchmark model for the titanic competition and updated the file's content manually and uploaded again to see if it's cached. The scores returned by Kaggle were different in the two submissions. So basically, it's not an issue.